docs/api_design.md

```diff
@@ -0,0 +1,69 @@
+# API Design
+
+> Warning: This is work in progress and APIs are very likely to change.
+
+This is a working draft of the API design for **s2-api**.
+
+## Anatomy
+
+```
+s2_api
+|-- Bot
+    |-- Action
+    |-- `tick()`
+|-- Connection
+|-- Engine
+    |-- GameState
+    |-- `run()`
+|-- Remote
+    |-- Controller
+        |-- `act()`, `observe()`, `ping()`, ...
+    |-- Protocol
+        |-- `send()` and `receive()`
+|-- SC2API
```

Comment:
I reckon a better name would be `raw_protobuf_api`, or something along those lines. Naming it *almost* the same as the crate will be confusing, plus we want to put emphasis on the fact that it's contains raw protobuf stuff and shouldn't be accessible by end users... Thoughts?